neighsyncd/neighsyncd.cpp

```diff
@@ -29,10 +48,24 @@ int main(int argc, char **argv)
             netlink.dumpRequest(RTM_GETNEIGH);
 
             s.addSelectable(&netlink);
+            if (sync.m_warmStartInProgress)
+            {
+                sync.readTableToMap();
+                WarmStart::setWarmStartState("neighsyncd", WarmStart::RESTORED);
+                sync.startReconcileTimer();
+            }
             while (true)
             {
                 Selectable *temps;
-                s.select(&temps);
+                s.select(&temps, SELECT_TIMEOUT);
```

Comment:
I would suggest to make use of SelectableTimer class by pushing an instance of it into the Select/Epoll loop. This logic is unnecessarily stopping the main loop every second, even when warm-reboot feature is disabled.

Reply:
This is timeout setting for select function, which only takes effect if nothing is active in the file descriptor or no processing for the signals.  This won't impact the performance when warm-reboot feature is enabled or disabled.

Reply:
Ass discussed, i'm not expecting any performance hit, but this approach doesn't sound right. We shouldn't rely on a permanent 1-second timer when we only need it a single time. In these cases we should add a 'selectable' timer within the 'select' object, and remove it when we don't need it anymore.